update_annotations_from_table.py

```diff
@@ -0,0 +1,182 @@
+"""This notebook updates all file annotations a project.
+
+There are four required file annotations for any assay file uploaded to Synapse.
+
+`UID` - a universal ID for the sample represented in the file
+`dataType`: `mRNA`, `miRNA`, `methylation`, etc.
+`fileType`: `bam`, `fastq`, etc.
+`fileSubType`: is `fileType`-specific; e.g., `bam` can have `mapped` or `unmapped`.
+
+This script uses the `dataType` field to get assay-specific files (using a query lookup dictionary `assayToQuery`),
+and then the `UID` to merge existing file annotations and the table metadata (which is obtained through a lookup
+dictionary `assayToMetadataTable`.
+
+"""
+
+import sys
+import pandas as pd
+import numpy
+import logging
+
+logger = logging.getLogger('Annotation Updater')
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+ch.setFormatter(formatter)
+logger.addHandler(ch)
+
+from multiprocessing.dummy import Pool
+mp = Pool(10)
+
+import synapseclient
+
+syn = synapseclient.Synapse(skip_checks=True)
+syn.login(silent=True)
+
+# Taken from https://github.com/larssono/Analysis_helpers/blob/master/synapseHelpers.py
+# Should probably be in synapseclient.synapseutils
+SYNAPSE_PROPERTIES = ['benefactorId', 'nodeType', 'concreteType', 'createdByPrincipalId',
```

Comment:
Yepp.  I will add this and the other items that came from SynapseHelpers in the next release.